### PR TITLE
update subscription & unit test with test case

### PR DESCRIPTION
**Pull request by ankitdas13**
for edit :  https://github.com/razorpay/razorpay-node/blob/master/documents/subscription.md#update-particular-subscription
Link : https://razorpay.com/docs/api/payments/subscriptions/#update-a-subscription

for pendingUpdate :  https://razorpay.com/docs/api/payments/subscriptions/#fetch-details-of-a-pending-update
Link : https://github.com/razorpay/razorpay-node/blob/master/documents/subscription.md#fetch-details-of-pending-update

for  cancelScheduledChanges  : https://github.com/razorpay/razorpay-node/blob/master/documents/subscription.md#cancel-a-update
Link : https://razorpay.com/docs/api/payments/subscriptions/#cancel-an-update

for pause : https://github.com/razorpay/razorpay-node/blob/master/documents/subscription.md#pause-a-subscription
Link : https://razorpay.com/docs/api/payments/subscriptions/#pause-a-subscription

for resume : https://github.com/razorpay/razorpay-node/blob/master/documents/subscription.md#resume-a-subscription
Link : https://razorpay.com/docs/api/payments/subscriptions/#resume-a-subscription

for deleteOffer : https://github.com/razorpay/razorpay-node/blob/master/documents/subscription.md#delete-offer-linked-to-a-subscription
Link : https://razorpay.com/docs/payments/subscriptions/offers/delete/#delete-an-offer-linked-to-a-subscription-using

### CODE REVIEW

**Review by sofiya02 — COMMENTED** [review]
Readme LGTM

File: lib/razorpay/request.rb [hunk 1]
```diff
@@ -47,10 +47,6 @@ def delete(url)
       request :delete, "/#{@entity_name}/#{url}"
     end
 
-    def delete(url)
-      request :delete, "/#{@entity_name}/#{url}"
-    end
-
```
Comment (sonyy-m) [inline lib/razorpay/request.rb:53]: why the delete request was removed?

**Review by sonyy-m — CHANGES_REQUESTED** [review]
Fix indentation and add newline at end of file in all json files

**Review by sonyy-m — APPROVED** [review]
LGTM

**Pull request merged**